src/matching.rs

```diff
@@ -101,49 +86,46 @@ fn parse_input_data(
     let mut node_id_to_caller_nodes: HashMap<String, HashSet<String>> = HashMap::new();
     let mut label_to_node_ids: HashMap<String, HashSet<String>> = HashMap::new();
 
-    let cg_file = File::open(callgraph_filename).unwrap();
-    for line in io::BufReader::new(cg_file).lines() {
-        if let Ok(contents) = line {
-            if contents.find("->").is_none() {
-                // found a new node!
-                for cap in node_re.captures_iter(&contents) {
-                    let node_id = cap[1].to_string();
-                    let full_label = cap[2].to_string();
-                    node_id_to_full_label.insert(node_id.clone(), full_label.clone());
-                    let short_label = simplify_trait_paths(full_label.clone());
-                    if let Some(node_ids) = label_to_node_ids.get_mut(&short_label) {
-                        node_ids.insert(node_id.to_string());
-                    } else {
-                        let mut hs = HashSet::new();
-                        hs.insert(node_id);
-                        label_to_node_ids.insert(short_label, hs);
-                    }
+    for contents in demangled_callgraph_lines {
+        if contents.find("->").is_none() {
+            // found a new node!
+            for cap in node_re.captures_iter(&contents) {
+                let node_id = cap[1].to_string();
+                let full_label = cap[2].to_string();
+                node_id_to_full_label.insert(node_id.clone(), full_label.clone());
+                let short_label = simplify_trait_paths(full_label.clone());
+                if let Some(node_ids) = label_to_node_ids.get_mut(&short_label) {
+                    node_ids.insert(node_id.to_string());
+                } else {
+                    let mut hs = HashSet::new();
+                    hs.insert(node_id);
+                    label_to_node_ids.insert(short_label, hs);
                 }
-            } else {
-                // found a new edge!
-                for cap in edge_re.captures_iter(&contents) {
-                    match node_id_to_caller_nodes.get_mut(&cap[2].to_string()) {
-                        Some(set) => {set.insert(cap[1].to_string());},
-                        None => {
-                            let mut set : HashSet<String> = HashSet::new();
-                            set.insert((&cap[1]).to_string());
-                            node_id_to_caller_nodes.insert(cap[2].to_string(), set);}
+            }
+        } else {
+            // found a new edge!
+            for cap in edge_re.captures_iter(&contents) {
```

Comment:
not sure what cap is here completely but maybe we can unpack it like `for (_, node_id, full_label) in ...`? (same above)

Reply:
those are captures from the regex- rust doesn't support that pattern match for capture object (as far as i can tell) but if you have another idea I'm open to it!

Reply:
Ahh; not aware of a better way then :P